apollo-normalized-cache/src/commonMain/kotlin/com/apollographql/apollo/cache/normalized/internal/CacheMapBuilder.kt

```diff
@@ -0,0 +1,116 @@
+package com.apollographql.apollo.cache.normalized.internal
+
+import com.apollographql.apollo.api.Operation
+import com.apollographql.apollo.api.ResponseField
+import com.apollographql.apollo.cache.CacheHeaders
+import com.apollographql.apollo.cache.normalized.CacheKey
+import com.apollographql.apollo.cache.normalized.CacheKeyResolver
+import com.apollographql.apollo.cache.normalized.CacheReference
+
+class CacheMapBuilder(
```

Comment:
Also curious if different name like `CacheBatchReader` / `CacheBufferBuilder` / `CacheBatchResolver` / `CacheRecordBatchReader` will be more descriptive the intention of this class?

Reply:
I'd like to keep `Buffer` for okio `Buffers` and trying to limit the use of `Resolver` as it's heavily used in the GraphQL ecosystem and I'm not sure we would use it like other frameworks would so that could be surprising. 

Between `CacheBatchReader` and `CacheMapBuilder`, it depends if we want to highlight the process (batch reading) or the result (a Map). Both work for me.

Reply:
I think we should high light this so we won't  be confused what this class for. `CacheBatchReader` probably should be ok.